coldfront/core/allocation/views_/secure_dir_views.py

```diff
@@ -757,3 +770,869 @@ def post(self, request, *args, **kwargs):
+        return context
+
+    def get_initial(self):
+        initial = super().get_initial()
+        rdm_consultation = self.request_obj.state['rdm_consultation']
+        initial['status'] = rdm_consultation['status']
+        initial['justification'] = rdm_consultation['justification']
+        return initial
+
+    def get_success_url(self):
+        return reverse(
+            'secure-dir-request-detail',
+            kwargs={'pk': self.kwargs.get('pk')})
+
+
+class SecureDirRequestReviewMOUView(LoginRequiredMixin,
+                                    UserPassesTestMixin,
+                                    SecureDirRequestMixin,
+                                    FormView):
+    form_class = SecureDirReviewStatusForm
+    template_name = (
+        'secure_dir/secure_dir_request/secure_dir_mou.html')
+    login_url = '/'
+
+    def test_func(self):
+        """UserPassesTestMixin tests."""
+        if self.request.user.is_superuser:
+            return True
+        message = 'You do not have permission to view the previous page.'
+        messages.error(self.request, message)
+        return False
+
+    def dispatch(self, request, *args, **kwargs):
+        pk = self.kwargs.get('pk')
+        self.set_request_obj(pk=pk)
+        redirect = self.redirect_if_disallowed_status(request)
+        if redirect is not None:
+            return redirect
+        return super().dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        form_data = form.cleaned_data
+        status = form_data['status']
+        justification = form_data['justification']
+        timestamp = utc_now_offset_aware().isoformat()
+        self.request_obj.state['mou'] = {
+            'status': status,
+            'justification': justification,
+            'timestamp': timestamp,
+        }
+        self.request_obj.status = secure_dir_request_state_status(self.request_obj)
+        self.request_obj.save()
+
+        if status == 'Denied':
+            runner = SecureDirRequestDenialRunner(self.request_obj)
+            runner.run()
+
+        message = (
+            f'MOU status for the secure directory request of project '
+            f'{self.request_obj.project.pk} has been set to {status}.')
+        messages.success(self.request, message)
+
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['secure_dir_request'] = self.request_obj
+        return context
+
+    def get_initial(self):
+        initial = super().get_initial()
+        mou = self.request_obj.state['mou']
+        initial['status'] = mou['status']
+        initial['justification'] = mou['justification']
+        return initial
+
+    def get_success_url(self):
+        return reverse(
+            'secure-dir-request-detail',
+            kwargs={'pk': self.kwargs.get('pk')})
+
+
+class SecureDirRequestReviewSetupView(LoginRequiredMixin,
```

Comment:
I know we said the user should provide a suggested name for the directory, but:

1. They could've put something inappropriate, or
2. Though unlikely, the name could have been taken by someone else (we're not actually checking for this during the request, I don't think).

There should be a field (initially containing the given name) so that the admin can overwrite it.